src/main/java/org/zalando/nakadi/repository/kafka/KafkaCursor.java

```diff
@@ -33,12 +34,14 @@ public KafkaCursor addOffset(final long toAdd) {
         return new KafkaCursor(topic, partition, offset + toAdd);
     }
 
-    public NakadiCursor toNakadiCursor() {
-        return new NakadiCursor(topic, toNakadiPartition(partition), toNakadiOffset(offset));
+    public NakadiCursor toNakadiCursor(final Timeline timeline) {
+        return new NakadiCursor(timeline,
+                toNakadiPartition(partition),
+                toNakadiOffset(offset));
     }
 
     public static String toNakadiOffset(final long offset) {
-        return String.valueOf(offset);
+        return offset >= 0 ? String.format("%018d", offset) : String.valueOf(offset);
```

Comment:
You do this `String.format("%018d", offset)` in many places. Maybe it will make sense to make some helper static method that does it with respect to the length constant we have for cursors with leading zeroes.